src/hooks/ExtensionProvider.ts

```diff
@@ -0,0 +1,24 @@
+import { createElement, createContext, FC, useContext } from "react";
+
+interface ExtensionProviderProps {
+  id?: string;
+  port?: browser.runtime.Port;
+}
+
+const context = createContext<ExtensionProviderProps>({
+  id: undefined,
+  port: undefined,
+});
+const { Provider } = context;
+
+export const ExtensionProvider: FC<ExtensionProviderProps> = ({
+  id,
+  children,
+  port,
+}) => {
+  return createElement(Provider, { value: { id, port }, children });
```

Comment:
`port` is added but not used yet, this is setting up the pipeline for subscriptions to work.